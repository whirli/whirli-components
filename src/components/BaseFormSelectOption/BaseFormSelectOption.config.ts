// @ts-ignore
import { PropType } from '@composition';
import * as PropHelpers from '@whirli-components/helpers/props';
import { ComponentStyles, ComponentProps } from '@whirli-components/@types/components';

// Local config
import config from '@whirli-local/components/BaseFormSelectOption/BaseFormSelectOption.config';

// Package config
import { PropKeys } from './BaseFormSelectOption.constants';

export const ConfigStyles: ComponentStyles = {
  ...config.styles,
};

export const ConfigProps: ComponentProps = {
  [PropKeys.SELECTED]: {
    required: true,
    type: [String, Number] as PropType<string | number>,
  },
  [PropKeys.VALUE]: {
    required: true,
    type: [String, Number] as PropType<string | number>,
  },
  ...config.props,
};
