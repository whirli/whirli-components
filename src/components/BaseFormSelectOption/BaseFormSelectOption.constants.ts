export const PropKeys: Record<string, string> = {
  SELECTED: 'selected',
  VALUE: 'value',
};

export const PropValues: Record<string, string[]> = {};
