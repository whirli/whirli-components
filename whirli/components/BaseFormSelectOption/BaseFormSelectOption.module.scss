@import '@whirli-components/components/BaseFormSelectOption/BaseFormSelectOption.module.scss';

// Standard styles

// Breakpoint styles
$localBreakpointStyles: ();

@include createBreakpointClasses(map-merge($packageBreakpointStyles, $localBreakpointStyles));