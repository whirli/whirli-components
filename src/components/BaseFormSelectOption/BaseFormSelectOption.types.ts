import { PropKeys, PropValues } from './BaseFormSelectOption.constants';

// const propColor: string[]  = PropValues[PropKeys.COLOR];

// export type PropColor = typeof propColor[number];
