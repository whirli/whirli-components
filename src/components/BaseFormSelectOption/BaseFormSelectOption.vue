<template>
  <option :selected="isSelected" :value="props.value"><slot /></option>
</template>

<script setup lang="ts">
import { computed } from '@composition';
import type { ComputedRef } from '@composition';

// Data
import { ConfigProps } from './BaseFormSelectOption.config';

// Types
import { Props } from '@whirli-components/@types/props';

// @ts-ignore
const props: Props = defineProps(ConfigProps);

const isSelected: ComputedRef<boolean> = computed(() => props.value === props.selected);
</script>
