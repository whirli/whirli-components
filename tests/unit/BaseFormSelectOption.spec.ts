import { mount } from '@vue/test-utils';
import BaseFormSelectOption from '@/components/BaseFormSelectOption/BaseFormSelectOption.vue';

let wrapper: any;
const value = 'hello-world';
const selected = 'hello-world';
const slot = 'Hello';

describe('BaseFormSelectOption.vue', () => {
  beforeEach(() => {
    wrapper = mount(BaseFormSelectOption, {
      slots: {
        default: slot,
      },
      props: {
        value,
        selected,
      },
    });
  });

  it('renders the component', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('displays the correct text', () => {
    expect(wrapper.findComponent(BaseFormSelectOption).text()).toMatch(slot);
  });

  it('has the correct value', () => {
    expect(wrapper.findComponent(BaseFormSelectOption).attributes('value')).toMatch(value);
  });
});
