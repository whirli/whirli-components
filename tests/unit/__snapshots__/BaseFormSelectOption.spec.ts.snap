// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseFormSelectOption.vue renders the component 1`] = `
<option
  value="hello-world"
>
  
  Hello
  
</option>
`;
